### ISSUE TITLE
BTable.vue no longer has currect typing for items is cell slots.

### ISSUE THREAD

**Issue by danielvy** [bootstrap-vue-next/bootstrap-vue-next#2034]
### Describe the bug

In `BTable.vue`, template cell slot scope no longer has the current type of the item and now defaults to `any`.

This was introduced in v0.23.0, commit: 3098cf2 in `BTable.vue`.

Previously, computedFields retrained the generic type T:
```
const computedFields = computed<TableFieldRaw<T>[]>(() =>
```

Now after the change, it is erased and defaults to `T = any` by `TableField`:
```
const computedFields = computed<TableField[]>(() =>
```

### Reproduction

https://stackblitz.com/edit/github-zbvgut

### Used Package Manager

pnpm

### Validations

- [X] Have tested with the latest version. This is still alpha version and sometime things change rapidly.
- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by VividLemon** [comment]
I thought I fixed this. Is this still a problem?

**Comment by danielvy** [comment]
> I thought I fixed this. Is this still a problem?

I see now it was fixed after I opened the issue, in v0.23.4

You can close the issue, thanks!
